index.d.ts

```diff
@@ -36,7 +36,7 @@ export namespace auth {
    * with it occurs.
    * - A string: Appropriate for simple value passing (e.g., providing OTP manually for logging in)
    */
-  type AuthEventHandler = string | (() => string | Promise<string>);
+  type AuthEventHandler = string | (() => string | Promise<string | void>);
```

Comment:
my bad, i totally forgot you are passing the tokens to the refresh handler. This isn't a pretty change but it'll do for now:

` type AuthEventHandler = string | (() => string | Promise<string>) | ((tokens: AuthTokens) => Promise<void>);`